lib/services/api_handler.dart

```
@@ -23,49 +24,60 @@ Future<void> callAPI({
-
-        for (String id in wm.references!.identifier) {
-          // check all warnings for references
-          for (WarnMessage alWm in warnings) {
-            debugPrint(alWm.identifier);
-            if (alWm.identifier.compareTo(id) == 0) {
-              // set flag to true to hide the previous alert in the overview
-              alWm.hideWarningBecauseThereIsANewerVersion =
-                  true; //@todo move to better location
+      // set flag for updated alerts
+      for (WarnMessage wm in warnings) {
+        if (wm.references != null) {
+          // the alert contains a reference, so it is an update of an previous alert
+          // we search for the alert and add it to the update thread
+
+          for (String id in wm.references!.identifier) {
+            // check all warnings for references
+            for (WarnMessage alWm in warnings) {
+              debugPrint(alWm.identifier);
+              if (alWm.identifier.compareTo(id) == 0) {
+                // set flag to true to hide the previous alert in the overview
+                alWm.hideWarningBecauseThereIsANewerVersion =
+                    true; //@todo move to better location
+              }
             }
           }
         }
       }
-    }
 
-    var updatedPlace = Place.withWarnings(
-      boundingBox: place.boundingBox,
-      subscriptionId: place.subscriptionId,
-      name: place.name,
-      warnings: warnings,
-      eTag: place.eTag,
-    );
-    placesWithWarningsList.add(updatedPlace);
+      var updatedPlace = Place.withWarnings(
+        boundingBox: place.boundingBox,
+        subscriptionId: place.subscriptionId,
+        name: place.name,
+        warnings: warnings,
+        eTag: place.eTag,
+      );
+      placesWithWarningsList.add(updatedPlace);
+    } on InvalidSubscriptionError {
+      //@TODO(Nucleus): show alert to user or automatically resubscribe to area
+      continue;
+    } catch (e) {
```

Please never do this. _Always_ catch specific exceptions and _never_ do catch alls. This kind of error logging just leads to ignoring errors really and doesn't accomplish much. It's better to know something is going wrong because at least then we get bug reports which we can act upon.

---

true, we can set `appState.error = true`. This results in an error message showing the user the log entry. How would you suggest handling the different exception types? There is not much we can do except write it to the log file and inform the user about the error. I don't see the point of writing down all possible exception types just to do the same thing. Also, we risk forgetting one, which results in an even worse situation, right?